projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentServiceV2.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.deploy;
+
+import com.vmware.taurus.controlplane.model.data.DataJobResources;
+import com.vmware.taurus.exception.*;
+import com.vmware.taurus.service.model.*;
+import com.vmware.taurus.service.notification.NotificationContent;
+import com.vmware.taurus.service.repository.JobsRepository;
+import io.kubernetes.client.openapi.ApiException;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * CRUD operations for Versatile Data Kit deployments on kubernetes.
+ *
+ * <p>This class is in a transition from operating against Kubernetes to operating against the
+ * database. Currently, only the enabled flag is stored in the database. Eventually, all deployment
+ * information will be stored (and retrieved) from a database from a dedicated table.
+ */
+@Service
+@RequiredArgsConstructor
+public class DeploymentServiceV2 {
+  private static final Logger log = LoggerFactory.getLogger(DeploymentServiceV2.class);
+
+  private final DockerRegistryService dockerRegistryService;
+  private final DeploymentProgress deploymentProgress;
+  private final JobImageBuilder jobImageBuilder;
+  private final JobImageDeployerV2 jobImageDeployer;
+  private final JobsRepository jobsRepository;
+  private final SupportedPythonVersions supportedPythonVersions;
+
+  /**
+   * Updates or creates a Kubernetes CronJob based on the provided configuration.
+   *
+   * <p>This method takes a CronJob configuration and checks if a CronJob with the same name already
+   * exists in the Kubernetes cluster. If the CronJob exists, it will be updated with the new
+   * configuration; otherwise, a new CronJob will be created. The method ensures that the CronJob in
+   * the cluster matches the desired state specified in the configuration.
+   *
+   * @param dataJob The data job to update or create.
+   * @param sendNotification if it is true the method will send a notification to the end user.
+   * @param cronJobs list of actual Kubernetes CronJobs.
```

Comment:
And that seems slightly fishy to me. Like we are breaking encapsulation. Cron Jobs are "data/infra layer" and not service layer.

Reply:
Renamed to dataJobDeploymentNames